vanilla-framework/vanilla-framework
# Add parker stylesheet analysis tool to set benchmarks

## Done

- Added the [parker](https://github.com/katiefenn/parker) tool to `package.json`
- Added simple benchmark tests in `/gulp/parker.js` (NOTE: thresholds are not final. See https://github.com/vanilla-framework/vanilla-framework/issues/1858)
- Added a report script that displays a breakdown of benchmark results
- Removed old test file

## QA

- Pull code
- Run `./run test`
- Check that the `parker:test` gulp test is run and that all tests succeed
- Open `/gulp/parker.js` and change one or more of the thresholds to something that will fail (e.g. Stylesheet size threshold to 150000)
- Run `./run test` again and check that the test fails, and the console output is helpful
- Run `./run exec yarn report`
- Check that the `parker:report` gulp test is run and that the console displays a list of benchmark results
